MaceWindu/sourcemap-tools
# Updates for v2 release

- [Esprima](https://www.nuget.org/packages/Esprima): 2.0.2 -> 2.1.2
- [System.Text.Json](https://www.nuget.org/packages/System.Text.Json): 5.0.2 -> 6.0.2
- [Microsoft.SourceLink.GitHub](https://www.nuget.org/packages/Microsoft.SourceLink.GitHub): 1.0.0 -> 1.1.1
- [NUnit3TestAdapter](https://www.nuget.org/packages/NUnit3TestAdapter): 4.0.0 -> 4.2.1
- [Microsoft.NET.Test.Sdk](https://www.nuget.org/packages/Microsoft.NET.Test.Sdk): 16.10.0 -> 17.0.0
- dropped `netcoreapp2.1` testing target
- added `net6.0` testing target
- switch build server to `windows-2022`
- repository cleanup: `.gitignore` update, copyright years update, remove unused upstream files
- use C# 10
- enable and fix CAxxxx/IDExxxx analyzers
- apply `[SkipLocalsInit]`
- added readme to nuget
- bumped version to 2.0.0 due to minor public api refactorings:
  - seal classes that not expected to be extended
  - add some missing `ArgumentNullException` to public APIs
  - `DeminifyStackTrace` method has additional `bool preferSourceMapsSymbols` parameter from https://github.com/microsoft/sourcemap-toolkit/pull/86
- fix [#16](https://github.com/MaceWindu/sourcemap-tools/issues/16)
  - merged https://github.com/microsoft/sourcemap-toolkit/pull/88
  - merged https://github.com/microsoft/sourcemap-toolkit/pull/86
  - add public setters to `StackFrame.MethodName` and `StackFrame.FilePath`
